### PR TITLE
Updated ML definitions

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

All `Id` issues will be fixed in a separate pr.

Related: https://github.com/elastic/kibana/pull/83808#issuecomment-791479571
/cc @jgowdyelastic

### CODE REVIEW

File: output/typescript/types.ts [hunk 1]
```diff
@@ -6461,6 +6499,8 @@ export type ModelMemoryStatus = 'ok' | 'soft_limit' | 'hard_limit'
 
 export interface ModelPlotConfig {
   terms: Fields
```
Comment (jgowdyelastic) [inline output/typescript/types.ts:6501]: should be `string`

File: output/typescript/types.ts [hunk 2]
```diff
@@ -7697,6 +7742,7 @@ export interface PutDatafeedRequest extends RequestBase {
     chunking_config?: ChunkingConfig
     frequency?: Time
     indices?: Indices
+    indexes?: Indices
```
Comment (jgowdyelastic) [inline output/typescript/types.ts:7745]: `indices` and `indexes` should be `string[]`

File: output/typescript/types.ts [hunk 3]
```diff
@@ -11305,6 +11351,7 @@ export interface UpdateJobRequest extends RequestBase {
     model_snapshot_retention_days?: long
     renormalization_window_days?: long
     results_retention_days?: long
+    groups?: Array<string>
```
Comment (jgowdyelastic) [inline output/typescript/types.ts:11354]: `groups` doesn't belong here

**Comment by jgowdyelastic** [comment]
will it also be possible to update the version of es these types are generated against so `findFileStructure` is removed from `kibana.d.ts`?

**Comment by delvedor** [comment]
> will it also be possible to update the version of es these types are generated against so findFileStructure is removed from kibana.d.ts?

These types are not generated, but created and maintained by hand, if a type is wrong it should be updated directly here :)

**Comment by delvedor** [comment]
I've added the `text_structure.find_structure` as well, in a subsequent or I'll update all of our json spec files, which are out of date.

File: specification/specs/x_pack/text_structure/FindStructure.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+interface FindStructureRequest<TBody> {
+  query_parameters?: {
+    charset?: string
+    column_names?: string
+    delimiter?: string
+    explain?: boolean
+    format?: string
+    grok_pattern?: string
+    has_header_row?: boolean
+    lines_to_sample?: uint
```
Comment (delvedor) [inline specification/specs/x_pack/text_structure/FindStructure.ts:34]: @philkra 💪

**Review by jgowdyelastic — APPROVED** [review]
LGTM!

**Pull request merged**